### ISSUE TITLE
AVT 1 - Search Component has a DAP violation 

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#4263]
## Environment
macOS Mojave version 10.14.5
Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React
Voice Over
DAP 1.8.0.0

## Detailed Description
Run DAP on the Search Component and the following violation is present:
<img width="1250" alt="Screen Shot 2019-10-08 at 1 32 09 PM" src="https://user-images.githubusercontent.com/21676914/66423045-73fde800-e9d0-11e9-926b-263edda1e722.png">

**Comment by abbeyhrt** [comment]
This is a super quick fix of just removing the unnecessary `role="img"` on the icon itself, since all icons are aria-hidden by default. I'm going to include this in a PR coming soon that will address this issue and #4187
